sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/ReactorNettyClient.java

```diff
@@ -104,7 +112,14 @@ private ReactorNettyClient(reactor.netty.http.client.HttpClient httpClient, Func
     public final HttpClient proxy(Supplier<ProxyOptions> proxyOptionsSupplier) {
         return new ReactorNettyClient(this.httpClient, client -> client.tcpConfiguration(c -> {
             ProxyOptions options = proxyOptionsSupplier.get();
-            return c.proxy(ts -> ts.type(options.type().value()).address(options.address()));
+            ProxyProvider.Proxy nettyProxy;
+            switch (options.type()) {
+                case HTTP: nettyProxy = ProxyProvider.Proxy.HTTP; break;
+                case SOCKS4: nettyProxy = ProxyProvider.Proxy.SOCKS4; break;
+                case SOCKS5: nettyProxy = ProxyProvider.Proxy.SOCKS5; break;
+                default: nettyProxy = null;
```

Comment:
I may just be misunderstanding this, but should the default case ever happen?  Should this be exceptional?

Reply:
Yep, definitely true. I was hacking last night just to get things compiling and passing linters. There are a number of cases where I didn't do things correctly in here. This is more of a 'squint your eyes and think about if this is the right approach' draft PR for now.